docu/Concepts/TechnicalRequirements/BusinessEventProtocol.md

```diff
@@ -0,0 +1,86 @@
+| TxSendEvent                 | 40    | the user creates a transaction and sends it online                                                   |
+| TxSendRedeemEvent           | 41    | the user creates a transaction and sends it per redeem link                                          |
+| TxRepeateRedeemEvent        | 42    | the user recreates a redeem link of a still open transaction                                         |
+| TxCreationEvent             | 50    | the user receives a creation transaction for his confirmed contribution                              |
+| TxReceiveEvent              | 51    | the user receives a transaction from an other user and posts the amount on his account               |
+| TxReceiveRedeemEvent        | 52    | the user activates the redeem link and receives the transaction and posts the amount on his account  |
+| ContribCreateEvent          | 60    | the user enters his contribution and asks for confirmation                                           |
+| ContribConfirmEvent         | 61    | the user confirms a contribution of an other user (for future multi confirmation from several users) |
+|                             |       |                                                                                                      |
+
+
+
+## EventProtocol - Entity
+
+The business events will be stored in database in the new table `EventProtocol`. The tabel will have the following attributes:
+
+| Attribute    | Type      | Description                                                                                      |
+| ------------ | --------- | ------------------------------------------------------------------------------------------------ |
+| id           | int       | technical unique key (from db sequence)                                                          |
+| type         | enum      | type of event                                                                                    |
+| createdAt    | timestamp | timestamp the event occurs (not the time of writing)                                             |
+| userID       | string    | the user ID, who invokes the event                                                               |
+| XuserID      | string    | the cross user ID, who is involved in the process like a tx-sender, contrib-receiver, ...        |
+| XcommunityID | string    | the cross community ID, which is involved in the process like a tx-sender, contrib-receiver, ... |
+| txID         | int       | the technical key of the transaction, which triggers the event                                   |
+| contribID    | int       | the technical key of the contribution, which triggers the event                                  |
+| amount       | digital   | the amount of gradido transferred by transaction, creation or redeem                             |
```

Comment:
we might also want a generic `reference` or `refID` field, which holds something like the `transactionID` or similar.

Reply:
the transactionID exists in form of the attribut txID.
for sure one or more general usable attribute could be added.
For the predefined list of event types all relevant values can be stored in the event protocol table